### ISSUE TITLE
Days are one day off if reselect date

### ISSUE THREAD

**Issue by ciis0** [brianblakely/nodep-date-input-polyfill#16]
Hi,

I think I experience similar issues as #6 and #13.

The concrete problem is that the Picker results in a wrong date, for me -2h off the date selected (so displayed as one day off). I think it's a time zone issue, as I'm UTC+1, currently with DST, so that matches perfectly (UTC+1 w/ DST is +1h so UTC is -2h).

I did some "debugging" (added some `console.log`s) and discovered your `Picker.absoluteDate` is called w/ `valueAsDate` when you open it (clicking into a polyfilled date input).

So the problem seems to be that the `valueAsDate` is already UTC (was stored as UTC previously) and `absoluteDate` makes it UTC-x (UTC-2 for me).

Steps to reproduce (at least for FF 45.2.0 (ESR)):

0. Have a polyfilled, empty `<input type="date">`
0. Click into it (this calls `Picker.absoulteDate` twice with `null`)
0. select any date, i.e. Aug 8 2016
0. Exit ´&lt;input>´ by clicking somewhere else (but stay on the page ;D)
0. Re-Enter (this calls `Picker.absoluteDate` twice with the selected date, i.e. Aug 8 2016 and returns Aug 7 2016 22:00h!)
0. Select any date, i.e. again Aug 8 2016
0. Be impressed input value now is Aug 7 2016!

For me it seems to work if I disable `Picker.absoluteDate` (so it returns the date w/o touching it), but I think you had your thoughts about the function.

tl;dr
Reselecting a date results in a date "one day off", because it UTC-fies already UTC-fied `valueAsDate` w/ `Picker.absoulteDate`.

Thank you.

**Comment by brianblakely** [comment]
Thanks for the detailed report

**Comment by christoph2schulz** [comment]
I'm currently unable to open a pullrequest or similar, but I've attached my changes as a patch that seem to work.

[nodep-date-input-polyfill-master.patch.txt](https://github.com/brianblakely/nodep-date-input-polyfill/files/430010/nodep-date-input-polyfill-master.patch.txt)

**Comment by chocorexic** [comment]
I had a play with nodep-date-input-polyfill last week and noticed the same issue in firefox and safari. I thought it was a problem with the browsers at first as there wasn't a problem with chrome, but that's because it does not use the polyfill since it supports input[type=date] natively right? 

I wanted to try the fix @c-2-s suggested but with my current setup I can't use the un-minified js as the build process complains about an unexpected string token from the import statements (for the css).

Will the offset date picked problem be fixed soon?

**Comment by brianblakely** [comment]
Goal is to deploy the patch today. 






On Wed, Aug 24, 2016 at 6:37 AM -0400, "Kit" <notifications@github.com> wrote:












I had a play with nodep-date-input-polyfill last week and noticed the same issue in firefox and safari. I thought it was a problem with the browsers at first as there wasn't a problem with chrome, but that's because it does not use the polyfill since it supports input[type=date] natively right? 



I wanted to try the fix @c-2-s suggested but with my current setup I can't use the un-minified js as the build process complains about an unexpected string token from the import statements (for the css).



Will the offset date picked problem be fixed soon?

**Comment by brianblakely** [comment]
Version 4.0.0 has been deployed.  The changes have been tested in 4 timezones, and the implementation now matches the native behavior in every way that we could muster.

This being a major version release, backwards compatibility may be broken.  Read the [usage notes](https://github.com/brianblakely/nodep-date-input-polyfill#usage-notes) before updating.

Thank you very kindly for the bug reports!  If you find anything else that's messed up, please keep 'em coming.
